src/crypto/openssl/rsa_key_pair.cpp

```diff
@@ -39,6 +39,40 @@ namespace crypto
     }
   }
 
+  RSAKeyPair_OpenSSL::RSAKeyPair_OpenSSL(const JsonWebKeyRSAPrivate& jwk)
+  {
+    auto rsa = RSAPublicKey_OpenSSL::rsa_public_from_jwk(jwk);
+
+    Unique_BIGNUM d, p, q, dp, dq, qi;
+    auto d_raw = raw_from_b64url(jwk.d);
+    auto p_raw = raw_from_b64url(jwk.p);
+    auto q_raw = raw_from_b64url(jwk.q);
+    auto dp_raw = raw_from_b64url(jwk.dp);
+    auto dq_raw = raw_from_b64url(jwk.dq);
+    auto qi_raw = raw_from_b64url(jwk.qi);
+
+    BN_bin2bn(d_raw.data(), d_raw.size(), d);
+    BN_bin2bn(p_raw.data(), p_raw.size(), p);
+    BN_bin2bn(q_raw.data(), q_raw.size(), q);
+    BN_bin2bn(dp_raw.data(), dp_raw.size(), dp);
+    BN_bin2bn(dq_raw.data(), dq_raw.size(), dq);
+    BN_bin2bn(qi_raw.data(), qi_raw.size(), qi);
+
+    CHECK1(RSA_set0_key(rsa, nullptr, nullptr, d));
+    CHECK1(RSA_set0_factors(rsa, p, q));
```

Comment:
We need to put `d.release()` above this line?
In case this line throws, the content of d is freed by `RSA_set0_key` and `Unique_BIGNUM` tries to free it again.